lua/resession/init.lua

```diff
@@ -462,8 +466,11 @@ M.load = function(name, opts)
       curwin = win
     end
   end
-  -- This can be nil if we saved a session in a window with an unsupported buffer
-  if curwin then
+  -- `curwin` can be nil if we saved a session in a window with an unsupported buffer.
+  -- Also check that all buffers loaded with success, and if not, give up on setting the window.
+  -- For some reason, without this check nvim crashes, e.g. if some of the loaded files have swap files.
+  -- This might have to do with the 'textlock' restriction, but I'm not sure.
+  if curwin and bufload_all_success then
```

Comment:
Oddly, I can't reproduce this problem in the tests/with the minimal init, and on my setup it happens only occasionally. So might have to do with a different plugin. But perhaps we can play it safe for now and keep this check.